metaflow/runner/deployer.py

```diff
@@ -271,6 +271,21 @@ class DeployedFlow(object):
 
     def __init__(self, deployer: "DeployerImpl"):
         self.deployer = deployer
+        self.name = self.deployer.name
+        self.flow_name = self.deployer.flow_name
+        self.metadata = self.deployer.metadata
+
+    @staticmethod
+    def from_deployment(
+        identifier: str,
+        metadata: str = None,
+        impl: str = "argo-workflows",
+    ):
+        if impl == "argo-workflows":  # TODO: use a metaflow config variable for `impl`
+            from metaflow.plugins.argo.argo_workflows_deployer import from_deployment
```

Comment:
one option for getting rid of argo specific impl details in core would be to introduce a new extension point through `metaflow.plugins`, f.ex. 

```
DEPLOYER_DESC = [
    ("argo-workflows", ".argo.argo_workflows_deployer")
]
```
and then just looking up the correct implementation from the list

Reply:
could support extending this through extensions as well but that change can be a different PR. `from_deployment` needs to be well defined at that point too.

Reply:
my preference would be inject these methods `from_argo_workflows` etc directly in the `DeployedFlow` class and have `from_deployment` be simply an alias that does something like this: 

```
@classmethod
def from_deployment(cls, identifier: str, metadata: str = None, impl: str = "argo-workflows")
    return getattr(cls, "from_%s" % impl)(identifier, metadata)
```
(modulo a few things like the _ and -).

We could inject these methods exactly like what we do in the `Deployer.__init__` except do it in the metaclass for `DeployedFlow`. For that matter, we should also probably do that in the `Deployer` metaclass too (because right now, we reprocess the deployer impl every time we init the class and set variables on the class itself (which is kind of weird).

Reply:
Please check the simpler way here: https://github.com/Netflix/metaflow/pull/2043/commits/a014447b645c203148d044813642bf03af0aad3b

Reply:
Hmm, this won't work in all cases...

Reply:
https://github.com/Netflix/metaflow/pull/2043/commits/5ee9c544765c60e796eb3572885bee45abd49956 works well..

we can now do stuff like:

```py
identifier = "parameterflowproject.user.madhurtandon23gmail.com.parameterflow"
metadata = "service@https://metadata.dev-latest.outerbounds.xyz/p/default/"

df = DeployedFlow.from_argo_workflows_deployment(identifier=identifier, metadata=metadata)
triggered_run = df.trigger(beta={"madhur": "cool"}, myfile="/Users/madhur/Desktop/output.csv")
```